lib/srv/desktop/tdp/proto.go

```diff
@@ -1304,6 +1306,11 @@ func decodeSharedDirectoryWriteRequest(in byteReader) (SharedDirectoryWriteReque
 		return SharedDirectoryWriteRequest{}, trace.Wrap(err)
 	}
 
+	if writeDataLength > maxLen {
+		io.CopyN(io.Discard, in, int64(writeDataLength))
```

Comment:
So, it's clear that is intentional.
```suggestion
		_, _ = io.CopyN(io.Discard, in, int64(writeDataLength))
```